### PR TITLE
distro/rhel9: add support for azure-sap-rhui

### PR DESCRIPTION
**Pull request by ondrejbudai**
We don't have a snapshot yet, so the test repository is just download.eng.bos.redhat.com for now. Let's fix it when we have a snapshot.

Regarding firewalld and libcanberra-gtk2:

The 1st party image has firewalld, but doesn't have libcanberra-gtk2. This is actually backwards that what is in out EC2 SAP images, so I had to do some shuffling to keep the behaviour in EC2, but have it different for Azure.

### CODE REVIEW

**Review by achilleas-k — DISMISSED** [review]

**Comment by ondrejbudai** [comment]
Forgot to update rhel-9 repos.

File: pkg/distro/rhel9/distro.go [hunk 1]
```diff
@@ -460,6 +460,8 @@ func newDistro(name string, minor int) *distribution {
 		x86_64.addImageTypes(azureX64Platform, azureRhuiImgType, azureByosImgType)
 		aarch64.addImageTypes(azureAarch64Platform, azureRhuiImgType, azureByosImgType)
 
+		x86_64.addImageTypes(azureX64Platform, azureSapRhuiImgType(rd))
```
Comment (thozza) [inline pkg/distro/rhel9/distro.go:463]: nitpick: you could have just extended the line 460

**Review by achilleas-k — APPROVED** [review]

**Review by thozza — APPROVED** [review]
LGTM

File: pkg/distro/rhel9/distro.go (hunk 1 above)
Comment (thozza) [reply pkg/distro/rhel9/distro.go:463]: :stuck_out_tongue_winking_eye:

**Pull request merged**